Amplify/Categories/DataStore/Subscribe/MutationEvent.swift

```diff
@@ -17,6 +17,8 @@ public struct MutationEvent: Model {
     public var json: String
     public var mutationType: String
     public var createdAt: Temporal.DateTime
+
+    @available(*, deprecated, message: "version field is deprecated")
```

Comment:
> I think for now, the safest way to do is mark it as deprecated field. If we remove the `version` field, shouldn't it be a major version change? For those customer who still using the mutation event version, the version will have the same value and meaning as before.

I'm not sure if marking as deprecated in this PR will provide any benefits to the customer or our codebase (without it cascading to additional changes). My question was more on whether we should do it or not, I am thinking now that deprecating `MutationEvent.version` is going to be a big effort out of scope of this PR, it is still the public facing contract for providing events through DataStore.observe API and other channels like the Hub. 

If it should be deprecated, we should make sure we know why we are doing this, perhaps customer doesn't care about the version. This would also be a big initiative to align the APIs across platforms, or effort to align with JS's API: https://docs.amplify.aws/javascript/build-a-backend/more-features/datastore/real-time/

The steps i'd suggest for this is..
- Keep MutationEvent.version not deprecated
- Treat MutationEvent's version as a snapshot of the version at the time it was hydrated. This means in OutgoingMutationQueue, we should stop modifying the snapshot version for the consecutive save scenario (remove MutationEvent.reconcilePendingMutationEventsVersion and calling code). We're getting the latest version in SyncToCloudMutationOperation anyways.
- Make sure all notify calls to the Hub and event propagation to DataStore.observe subscribers get a MutationEvent that contains the latest version
- There may be other areas that I missed, one way to find these is to remove the `version` from MutationEvent completely and see where the code depends on it. (Then revert removing the `version` field back of course)

Reply:
> Make sure all notify calls to the Hub and event propagation to DataStore.observe subscribers get a MutationEvent that contains the latest version

we looked over this and 
- OutgoingMutationQueue sends an OutboxMutationProcessedEvent which has a version coming from the remote model's version metadata, so it is the latest
- Reconcile path will send a MutationEvent that is reconstructed from the remote model's version as well. 
Both places use the latest version for creating the MutationEvent object for the public facing APis